tests/run_tests.sh

```diff
@@ -17,21 +19,23 @@ declare -a succesful_tests
 declare -a failed_tests
 test_count=0
 for test_file in $test_dir/*_test.yaml; do
+    test_filename=$(basename $test_file)
+    test_filename="${filename%.*}"
```

Comment:
You are storing values in the same variable, so the fist setting is never used. Is this intentional.

Reply:
Yes, as I don't want the filename extension. I could maybe consolidate to one command.

Reply:
Wait, I see the bug.